rasa/core/nlg/response.py

```diff
@@ -75,12 +75,26 @@ def _responses_for_utter_action(
         )
 
         if conditional_channel:
+            formatted_response_conditions = self._format_response_conditions(
+                conditional_channel[0].get(RESPONSE_CONDITION)
```

Comment:
Here I have an edge case that might cause an inconsistency with the selection of the first text variation in the conditional responses list. Say we have 2 text variations for the same utter action: variation 1 with condition that slot A is a certain value, variation 2 with condition that slot B is another value. If a user has both slots A and B filled with these values, both response variations become eligible conditional responses. In [this line](https://github.com/RasaHQ/rasa/blob/b2b5b73c10b7a7e5f3331a0f6887ae7d6f1782fd/rasa/core/nlg/response.py#L105), rasa will pick at random one of the text variations - this could be inconsistent with the debug log message, you would say selecting response with condition A, but actually the response with condition B was shown.

To avoid this inconsistency - I actually suggest moving the debug logging in `_random_response_for` method after this:
```
if suitable_responses:
        selected_response = np.random.choice(suitable_responses)
        condition = selected_response.get(RESPONSE_CONDITION)
        if condition:
            formatted_response_condition = ...
            logger.debug(f"Selecting response variation with conditions:"....)
        return selected_response
```

Hope this makes sense, let me know if it doesn't!

Maybe if you think it's a good idea to extract the lines between `selected response` assignment and return into its own logging method 🤔